openpype/modules/deadline/plugins/publish/submit_maya_deadline.py

```diff
@@ -272,37 +276,12 @@ def process(self, instance):
             self.deadline_url = instance.data.get("deadlineUrl")
         assert self.deadline_url, "Requires Deadline Webservice URL"
 
-        self._job_info = (
-            context.data["project_settings"].get(
-                "deadline", {}).get(
-                "publish", {}).get(
-                "MayaSubmitDeadline", {}).get(
-                "jobInfo", {})
-        )
+        # just using existing names from Setting
+        self._job_info = self.jobInfo
 
-        self._plugin_info = (
-            context.data["project_settings"].get(
-                "deadline", {}).get(
-                "publish", {}).get(
-                "MayaSubmitDeadline", {}).get(
-                "pluginInfo", {})
-        )
+        self._plugin_info = self.pluginInfo
 
-        self.limit_groups = (
-            context.data["project_settings"].get(
-                "deadline", {}).get(
-                "publish", {}).get(
-                "MayaSubmitDeadline", {}).get(
-                "limit", [])
-        )
-
-        self.group = (
-            context.data["project_settings"].get(
-                "deadline", {}).get(
-                "publish", {}).get(
-                "MayaSubmitDeadline", {}).get(
-                "group", "none")
-        )
+        self.limit_groups = self.limit
```

Comment:
Would be nicer if somehow we could refactor the rest of the code to just use the class variables directly instead of re-assigning them. I agree the `self.limit` name isn't very descriptive however.

If we'd just use the existing settings info we could however remove these three lines completely.

Should we refactor that too to cleanup some technical debt there? It might mean to also tweak the settings names however?

Reply:
Nah it's fine - just wondered that if we were doing cleanup anyway whether this would also be the moment to clean this up in settings now so that names match with the code better, etc.

Would be good to keep track of this and resolve it for v4 however. Especially since that's a completely new database we can do whatever we want it feels like.

Reply:
I don't think that how Settings are stored/parsed will be much different in v4. 
But I am all up for refactoring and clean up, but we don't have any tests for deadline publishing yet.

Reply:
> I don't think that how Settings are stored/parsed will be much different in v4.

The only reason not to rename now is that if a studio had a custom value set now that they'd lose their setting and had to re-apply their setting in the new setting attribute. With v4 I imagine with the new database that studios might need to set their settings anyway, OR we'll provide a 'convert v3 -> v4' script in which we can refactor names anyway? Not as nice, but it'd somewhat be a one time thing.

Reply:
I am wondering if we could utilize something like this:

https://github.com/pypeclub/OpenPype/blob/develop/openpype/modules/deadline/abstract_submit_deadline.py

in that file is list of all options deadline job can have. Maybe create Settings widget (entity) to easily add from that list with human readable description so it is unified across the hosts...? I'll create new issue about that to keep track.

Reply:
Oh yes baby. That sounds lovely.